src/python/pants/backend/python/lint/ruff/rules.py

```diff
@@ -95,15 +123,24 @@ async def ruff_lint(request: RuffLintRequest.Batch[RuffFieldSet, Any]) -> LintRe
         SourceFiles, SourceFilesRequest(field_set.source for field_set in request.elements)
     )
     result = await Get(
-        FallibleProcessResult, _RunRuffRequest(snapshot=source_files.snapshot, is_fix=False)
+        FallibleProcessResult, _RunRuffRequest(snapshot=source_files.snapshot, mode=RuffMode.LINT)
     )
     return LintResult.create(request, result)
 
 
+@rule(desc="Format with ruff", level=LogLevel.DEBUG)
+async def ruff_fmt(request: RuffFormatRequest.Batch, ruff: Ruff) -> FmtResult:
+    result = await Get(
+        FallibleProcessResult, _RunRuffRequest(snapshot=request.snapshot, mode=RuffMode.FORMAT)
+    )
+    return await FmtResult.create(request, result)
+
+
 def rules():
     return [
         *collect_rules(),
         *RuffFixRequest.rules(),
         *RuffLintRequest.rules(),
+        *RuffFormatRequest.rules(),
```

Comment:
I think it should be possible to add another flag, e.g. `--ruff-modes` that would allow you to set the modes that `ruff` will be invoked with. e.g. `--ruff-modes=["fmt"]` would only run the formatter, but specifying no modes would enable all three modes.

I'd be hesitant to land this PR before having that ability, since I think people will be surprised

Reply:
That's defined here: https://github.com/pantsbuild/pants/pull/20098/files#diff-0e7d34a4ac84a9ac63abd7b18c5759195d5061bab98777c5ddedaa3a366ec54dR56

Should just be a matter of adding `fix` to that list.

Reply:
Doesn't seem like there is a good solution for this problem per: https://pantsbuild.slack.com/archives/C0D7TNJHL/p1699642149609969?thread_ts=1699632752.483199&cid=C0D7TNJHL

so skipping for now.